### DISCUSSION TITLE
Add support for woocommerce_formatted_address_replacements

### DISCUSSION THREAD

**Discussion by artprojectgroup** [woocommerce/woocommerce#48959] [Checkout Flow]
When customising the address with `woocommerce_localisation_address_formats` and adding a new field it is not possible to replace it in checkout block as it does not support the `woocommerce_formatted_address_replacements` filter, it would be nice if support could be added or someone could explain me how to do it as I don't see a way to overwrite the addressTokens constant.

Thank you very much in advance.

Best regards.

**Comment by opr** [comment]
Hi @artprojectgroup are you trying to modify the way the address appears on the address card?

<img width="646" alt="image" src="https://github.com/woocommerce/woocommerce/assets/5656702/ca7de62a-5ece-475e-992a-1105f850f5b3">

Can you provide me an example of what you're trying to do and a reproducible use case? Support for address formats in the address card was added in https://github.com/woocommerce/woocommerce/pull/45852.

**Comment by artprojectgroup** [comment]
![image](https://github.com/woocommerce/woocommerce/assets/6569259/d01c7874-fb29-4e02-a655-19cab533e6d2)
I need to change {nif} to the value contained in the new NIF/CIF/NIE field:
![image](https://github.com/woocommerce/woocommerce/assets/6569259/616ee018-8746-4040-b948-51db5cd7c198)
Best regards.

**Comment by passatgt** [comment]
Same issue here, i previously included the VAT number and i displayed it after the company name between ():


```
		public static function add_vat_number_to_address( $formats ) {
			foreach($formats as $id => $format) {
				$formats[$id] = str_replace("{company}", "{company}{wc_szamlazz_adoszam}", $format);
			}
			return $formats;
		}

		public static function replace_vat_number_in_address( $replacements, $args ){
			$replacements['{wc_szamlazz_adoszam}'] = 'teszt';
			if(isset($args['wc_szamlazz_adoszam']) && !empty($args['wc_szamlazz_adoszam'])) {
				$replacements['{wc_szamlazz_adoszam}'] = ' ('.$args['wc_szamlazz_adoszam'].')';
			}
			return $replacements;
		}
```

**Reply by artprojectgroup** [reply]
What hooks are you using?

Best regards.

**Reply by passatgt** [reply]
```
add_filter( 'woocommerce_localisation_address_formats', array( __CLASS__, 'add_vat_number_to_address'));
add_filter( 'woocommerce_formatted_address_replacements', array( __CLASS__, 'replace_vat_number_in_address'), 10, 2);

```

**Reply by passatgt** [reply]
I added a is_checkout() condition in woocommerce_localisation_address_formats to keep it the original, so at least the empty placeholder is not visible anymore for now

**Reply by mojzesz777** [reply]
so this code works to finally be able to add the vat number field to the order fields?

**Reply by artprojectgroup** [reply]
Sorry but that code does not solve the problem. At the moment I use `if ( ! is_checkout() ) { ... }` in woocommerce_localisation_address_formats to prevent it from displaying.

Thank you very much anyway.

Best regards.
